tests/puppeteer/pages/BO/shopParameters/customerSettings/index.js

```diff
@@ -14,53 +14,39 @@ module.exports = class customerSettings extends BOBasePage {
     this.askForBirthDateLabel = 'label[for=\'form_general_ask_for_birthday_%TOGGLE\']';
+    this.enableB2BModeToggle = 'label[for=\'form_general_enable_b2b_mode_%TOGGLE\']';
     this.saveGeneralFormButton = `${this.generalForm} .card-footer button`;
   }
 
   /*
     Methods
   */
-
   /**
-   * Enable/disable redisplay cart at login
-   * @param toEnable, true to enable and false to disable
+   * Set option status
+   * @param option, option to enable or disable
+   * @param toEnable, value wanted
    * @return {Promise<string>}
    */
-  async setRedisplayCartAtLoginStatus(toEnable = true) {
-    await this.waitForSelectorAndClick(this.redisplayCartAtLoginLabel.replace('%TOGGLE', toEnable ? 1 : 0));
-    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
-    return this.getTextContent(this.alertSuccessBlock);
-  }
-
-  /**
-   * Enable/Disable send email after registration
-   * @param toEnable
-   * @returns {Promise<string>}
-   */
-  async setEmailAfterRegistrationStatus(toEnable = true) {
-    await this.waitForSelectorAndClick(this.sendEmailAfterRegistrationLabel.replace('%TOGGLE', toEnable ? 1 : 0));
-    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
-    return this.getTextContent(this.alertSuccessBlock);
-  }
-
-  /**
-   * Enable/Disable ask for birth date
-   * @param toEnable
-   * @returns {Promise<string|*>}
-   */
-  async setAskForBirthDate(toEnable = true) {
-    await this.waitForSelectorAndClick(this.askForBirthDateLabel.replace('%TOGGLE', toEnable ? 1 : 0));
-    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
-    return this.getTextContent(this.alertSuccessBlock);
-  }
-
-  /**
-   * Enable/Disable partner offer
-   * @param toEnable
-   * @returns {Promise<string>}
-   */
-  async setPartnerOfferStatus(toEnable = true){
-    await this.waitForSelectorAndClick(this.enablePartnerOfferLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+  async setOptionStatus(option, toEnable = true) {
+    switch (option) {
+      case 'B2B mode':
+        await this.waitForSelectorAndClick(this.enableB2BModeToggle.replace('%TOGGLE', toEnable ? 1 : 0));
+        break;
+      case 'Partner offers':
+        await this.waitForSelectorAndClick(this.enablePartnerOfferLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+        break;
+      case 'Ask for birth date':
+        await this.waitForSelectorAndClick(this.askForBirthDateLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+        break;
+      case 'Email after registration':
+        await this.waitForSelectorAndClick(this.sendEmailAfterRegistrationLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+        break;
+      case 'Redisplay cart at login':
+        await this.waitForSelectorAndClick(this.redisplayCartAtLoginLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+        break;
+      default:
+        throw new Error(`${option} was not found in the page`);
+    }
```

Comment:
```suggestion
  async setOptionStatus(selector, toEnable = true) {
    await this.waitForSelectorAndClick(selector.replace('%TOGGLE', toEnable ? 1 : 0));
```

Reply:
And then replaces every use of .setOptionsStatus with selector instead, the use of a switch there is the perfect contrary of a reusable component method!

Don't forget to throw an error if waitForSelectorAndClick doesn't find the selector (not if you put a selector which isn't in a switch case)

Reply:
as I said, we separate tests logic from pages logic, so we don't want to call the selector in test.

Reply:
The other option is to do it like this :

```js
  async setOptionStatus(selector, toEnable = true) {
    await this.waitForSelectorAndClick(selector.replace('%TOGGLE', toEnable ? 1 : 0));
    await this.clickAndWaitForNavigation(this.saveGeneralFormButton);
    return this.getTextContent(this.alertSuccessBlock);
  }
  
  setPartnerOfferStatus(toEnable = true) {
    return this.setOptionStatus(this.enablePartnerOfferLabel, toEnable);
  }
```

Reply:
@PierreRambaud what do you think ?

Myself, I think thats a good compromise even if it adds unwanted complexity, but if they wants to separate logics, sometimes it's needed..